File: Behavioural-Patterns/Observer/Java/Implementations/LogObserver.java
package Implementations;

import java.time.LocalTime;
import java.time.format.DateTimeFormatter;

import javax.swing.JTextArea;

import Interfaces.Observer;
import Interfaces.Subject;

public class LogObserver implements Observer {

    private final JTextArea logArea;

    public LogObserver(JTextArea jTextArea){
        this.logArea = jTextArea;
    }

    @Override
    public void update(Subject source) {
        if (source instanceof CustomButton) {
            CustomButton button = (CustomButton) source;
            String buttonText = button.getText();
            String timeStamp = LocalTime.now().format(DateTimeFormatter.ofPattern("HH:mm:ss"));
            
            logArea.append("[" + timeStamp + "] Received notification from: '" + buttonText + "' button.\n");
        }
    }
    
}

File: Behavioural-Patterns/Observer/C#/Observer/Implementations/LogObserver.cs
using System;
using System.Windows.Forms;

public class LogObserver : IObserver {

    private readonly TextBox logArea;

    public LogObserver(TextBox jTextArea) {
        this.logArea = jTextArea;
    }

    public void Update(Subject source) {
        if (source is CustomButton button) {
            String buttonText = button.Text;
            String timeStamp = DateTime.Now.ToString("HH:mm:ss");

            logArea.AppendText($"[{timeStamp}] Received notification from: '{buttonText}' button.{Environment.NewLine}");
        }
    }

}
